clang/lib/Sema/TreeTransform.h

```diff
@@ -15609,6 +15609,14 @@ TreeTransform<Derived>::TransformCXXFoldExpr(CXXFoldExpr *E) {
       return true;
   }
 
+  // When there's only one expansion, the parentheses can be safely eliminated
+  // to avoid any extra redundancy that may result in incorrect checks.
+  // For example, transforming (((args == 0) || ...)) into (args == 0)
+  // allows the omission of parentheses while ensuring precise representation
+  // and avoiding warnings regarding redundant parentheses.
+  if (*NumExpansions == 1 && !E->isTypeDependent())
```

Comment:
I don't think this is right, decltype isn't the only situation where this could be type dependent.  I think we really do have to settle for putting a 'bit' on `ParenExpr`.

Reply:
Since it's only known to occur with fold expressions, should it just be a bool, or would it be better to use an enum?

Reply:
I like doing enums, only because they are more descriptive, so I'd suggest one of those, but of size `uint8_t` on `ParenExpr`.  If we decide to add a different 'origin' later, that might be useful.